smartsim/entity/_new_ensemble.py

```diff
@@ -51,26 +52,35 @@ def __init__(
         exe: str | os.PathLike[str],
         exe_args: t.Sequence[str] | None = None,
         files: EntityFiles | None = None,
-        parameters: t.Mapping[str, t.Sequence[str]] | None = None,
+        file_parameters: t.Mapping[str, t.Sequence[str]] | None = None,
         permutation_strategy: str | strategies.TPermutationStrategy = "all_perm",
-        max_permutations: int = 0,
+        exe_arg_parameters: t.Mapping[str, t.Sequence[t.Sequence[str]]] | None = None,
+        max_permutations: int = 30,
         replicas: int = 1,
     ) -> None:
         self.name = name
         self.exe = os.fspath(exe)
         self.exe_args = list(exe_args) if exe_args else []
         self.files = copy.deepcopy(files) if files else EntityFiles()
-        self.parameters = dict(parameters) if parameters else {}
+        self.file_parameters = dict(file_parameters) if file_parameters else {}
         self.permutation_strategy = permutation_strategy
+        self.exe_arg_parameters = (
+            copy.deepcopy(exe_arg_parameters) if exe_arg_parameters else {}
+        )
         self.max_permutations = max_permutations
         self.replicas = replicas
 
     def _create_applications(self) -> tuple[Application, ...]:
         permutation_strategy = strategies.resolve(self.permutation_strategy)
-        permutations = permutation_strategy(self.parameters, self.max_permutations)
-        permutations = permutations if permutations else [{}]
-        permutations_ = itertools.chain.from_iterable(
-            itertools.repeat(permutation, self.replicas) for permutation in permutations
+        combinations = permutation_strategy(
+            self.file_parameters, self.exe_arg_parameters, self.max_permutations
+        )
+        combinations = combinations if combinations else [ParamSet({}, {})]
+        permutations_ = list(
```

Comment:
nit: no need to cast to list here if you are just planning on iterating over it:
```suggestion
        permutations_ = (
```